Incompatible with ESM module (node > 13.2)

Threads.js doesn't work with node ESM module, if `"type": "module"` is set in the `package.json` importing threads.js as a dependency, same if main file extension is `.mjs`.

Worker constructor fails with the following error:
```
UnhandledPromiseRejectionWarning: Error: Cannot find module 'file:/(...)/add.js'
Require stack:
- (...)/node_modules/threads/dist/master/implementation.node.js
- (...)/node_modules/threads/dist/master/implementation.js
- (...)/node_modules/threads/dist/master/index.js
- (...)/node_modules/threads/dist/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:976:15)
    at Function.resolve (internal/modules/cjs/helpers.js:78:19)
    at resolveScriptPath ((...)/node_modules/threads/dist/master/implementation.node.js:63:19)
    at new Worker ((...)/node_modules/threads/dist/master/implementation.node.js:74:40)
    at file:///(...)/test.js:4:35
    at file:///(...)/test.js:10:3
    at ModuleJob.run (internal/modules/esm/module_job.js:110:37)
    at async Loader.import (internal/modules/esm/loader.js:167:24)
``` 

Reproduce (requires node > 13.2 and package.json `"type":"module"`):
test.js
```
import threads from 'threads'

(async () => {
  const add = await threads.spawn(new threads.Worker("./add.js"))
  const result = await add(1, 1)

  console.log("result:", result)

  await threads.Thread.terminate(add)
})()
```

add.js
```
import { expose }  from "threads/worker"

expose({
  add(a, b) {
    return a + b
  }
})
```

Note: adding extension to worker path or writing worker in commonJS with .cjs extension doesn't change anything.

PS: Thanks for this awesome library !

@andywer Yes, I removed my personal path from the stacktrace but using `ls` on the path returned by `Cannot find module 'file:/(...)/add.js'` is successful.

I'll push the test repo for reproduction ;)

I need to do the import as follows to avoid getting import error:
```
import threads from 'threads';
const Worker = threads.Worker;
const Pool = threads.Pool;
const spawn = threads.spawn;
```

And even after removing the *.mjs files I still get the following error:
```
Error: Cannot find module 'C:\<my_path>\node_modules\threads\dist\master\file:\C:\<my_path>\nodejs\tests\workers.js'
Require stack:
- C:\<my_path>\nodejs\node_modules\threads\dist\master\implementation.node.js
- C:\<my_path>\nodejs\node_modules\threads\dist\master\implementation.js
- C:\<my_path>\nodejs\node_modules\threads\dist\master\index.js
- C:\<my_path>\nodejs\node_modules\threads\dist\index.js
...
code: 'MODULE_NOT_FOUND',
```
tbh I reverted my project back to commonJS modules and it works fine.  I'm using node v14.2.0.

--- a/index.mjs
+++ b/index.mjs
@@ -1,0 +1,9 @@
+import Threads from "./dist/index.js"
+
+export const registerSerializer = Threads.registerSerializer
+export const spawn = Threads.spawn
+export const DefaultSerializer = Threads.DefaultSerializer
+export const Pool = Threads.Pool
+export const Thread = Threads.Thread
+export const Transfer = Threads.Transfer
+export const Worker = Threads.Worker

--- a/observable.mjs
+++ b/observable.mjs
@@ -1,1 +1,4 @@
-export * from "./dist-esm/observable"
+import Observables from "./dist/observable.js"
+
+export const Observable = Observables.Observable
+export const Subject = Observables.Subject

--- a/package.json
+++ b/package.json
@@ -4,6 +4,12 @@
   "description": "Web workers & worker threads as simple as a function call",
   "main": "dist/index.js",
   "module": "dist-esm/index.js",
+  "exports": {
+    ".": "./index.mjs",
+    "./observable": "./observable.mjs",
+    "./register": "./register.mjs",
+    "./worker": "./worker.mjs"
+  },
   "scripts": {
     "prebuild": "rimraf dist/ dist-esm/",
     "build": "run-p build:cjs build:es",

--- a/register.mjs
+++ b/register.mjs
@@ -1,1 +1,1 @@
-export * from "./dist-esm/master/register"
+import "./dist/master/register.js"

--- a/src/master/implementation.node.ts
+++ b/src/master/implementation.node.ts
@@ -64,7 +64,8 @@ function rebaseScriptPath(scriptPath: string, ignoreRegex: RegExp) {
     )
   })
 
-  const callerPath = parentCallSite ? parentCallSite.getFileName() : null
+  const rawCallerPath = parentCallSite ? parentCallSite.getFileName() : null
+  const callerPath = rawCallerPath ? rawCallerPath.replace(/^file:\//, "") : null
   const rebasedScriptPath = callerPath ? path.join(path.dirname(callerPath), scriptPath) : scriptPath
 
   return rebasedScriptPath

--- a/worker.mjs
+++ b/worker.mjs
@@ -1,1 +1,5 @@
-export * from "./dist-esm/worker/index"
+import WorkerContext from "./dist/worker/index.js"
+
+export const expose = WorkerContext.expose
+export const registerSerializer = WorkerContext.registerSerializer
+export const Transfer = WorkerContext.Transfer

